File: kaltura/api_v3/generator/output/dotNet/KalturaClient/Types/KalturaReportInputFilter.cs
using System;
using System.Xml;
using System.Collections.Generic;

namespace Kaltura
{
	public class KalturaReportInputFilter : KalturaObjectBase
	{
		#region Properties
		public int FromDate = Int32.MinValue;
		public int ToDate = Int32.MinValue;
		public string Keywords = null;
		public bool? SearchInTags = false;
		public bool? SearchInAdminTags = false;
		#endregion

		#region CTor
		public KalturaReportInputFilter()
		{
		}

		public KalturaReportInputFilter(XmlElement node)
		{
			foreach (XmlElement propertyNode in node.ChildNodes)
			{
				string txt = propertyNode.InnerText;
				switch (propertyNode.Name)
				{
					case "fromDate":
						this.FromDate = ParseInt(txt);
						continue;
					case "toDate":
						this.ToDate = ParseInt(txt);
						continue;
					case "keywords":
						this.Keywords = txt;
						continue;
					case "searchInTags":
						this.SearchInTags = ParseBool(txt);
						continue;
					case "searchInAdminTags":
						this.SearchInAdminTags = ParseBool(txt);
						continue;
				}
			}
		}
		#endregion

		#region Methods
		public new KalturaParams ToParams()
		{
			KalturaParams kparams = base.ToParams();
			kparams.AddIntIfNotNull("fromDate", this.FromDate);
			kparams.AddIntIfNotNull("toDate", this.ToDate);
			kparams.AddStringIfNotNull("keywords", this.Keywords);
			kparams.AddBoolIfNotNull("searchInTags", this.SearchInTags);
			kparams.AddBoolIfNotNull("searchInAdminTags", this.SearchInAdminTags);
			return kparams;
		}
		#endregion
	}
}

File: kaltura/api_v3/lib/types/report/KalturaReportInputFilter.php
<?php
/**
 * @package api
 * @subpackage objects
 */
class KalturaReportInputFilter extends KalturaObject 
{
	/**
	 * @var int
	 */
	public $fromDate;
	
	/**
	 * @var int
	 */
	public $toDate;
	
	/**
	 * 
	 * @var string
	 */
	public $keywords;
	
	/**
	 * @var bool
	 */
	public $searchInTags;	

	/**
	 * @var bool
	 */
	public $searchInAdminTags;	
	

	public function toReportsInputFilter ()
	{
		$reportInputFilter = new reportsInputFilter();
		$reportInputFilter->from_date = $this->fromDate;
		$reportInputFilter->to_date = $this->toDate;
		$reportInputFilter->keywords = $this->keywords;
		$reportInputFilter->search_in_tags= $this->searchInTags;
		$reportInputFilter->search_in_admin_tags= $this->searchInAdminTags;
		
		return $reportInputFilter;
	}
	
	public function fromReportsInputFilter ( reportsInputFilter $reportInputFilter )
	{
		$this->fromDate = $reportInputFilter->from_date ;
		$this->toDate = $reportInputFilter->to_date ;
		$this->keywords = $reportInputFilter->keywords ;
		$this->searchInTags = $reportInputFilter->search_in_tags ;
		$this->searchInAdminTags = $reportInputFilter->search_in_admin_tags ;
		
		return $this;
	}	
}